pkg/prompb/remote.pb.go

```diff
@@ -61,12 +61,27 @@ func (ReadRequest_ResponseType) EnumDescriptor() ([]byte, []int) {
 	return fileDescriptor_eefc82927d57d89b, []int{1, 0}
 }
 
+// ArrayPool contains arrays that are shared across Timeseries
+// This is to reduce the overhead of allocation of a lots of small arrays
+type ArrayPool struct {
+	labelsPool    []Label
+	samplesPool   []Sample
+	exemplarsPool []Exemplar
+}
+
 type WriteRequest struct {
 	Timeseries           []TimeSeries     `protobuf:"bytes,1,rep,name=timeseries,proto3" json:"timeseries"`
 	Metadata             []MetricMetadata `protobuf:"bytes,3,rep,name=metadata,proto3" json:"metadata"`
 	XXX_NoUnkeyedLiteral struct{}         `json:"-"`
 	XXX_unrecognized     []byte           `json:"-"`
 	XXX_sizecache        int32            `json:"-"`
+	XXX_arrayPool        *ArrayPool       // gogo proto forces us to have XXX_ prefix for unexported fields
```

Comment:
I wonder why not make this a struct embedding instead of a pointer. This makes it more friendly for GC 
```suggestion
	XXX_arrayPool        ArrayPool       // gogo proto forces us to have XXX_ prefix for unexported fields
```